autoscale/src/main/java/com/sequenceiq/periscope/controller/AutoScaleClusterCommonService.java

```diff
@@ -103,7 +108,11 @@ public Cluster setState(Long clusterId, StateJson stateJson) {
 
     public Cluster setAutoscaleState(Long clusterId, AutoscaleClusterState autoscaleState) {
         Cluster cluster = setAutoscaleState(clusterId, autoscaleState.isEnableAutoscaling());
-        return setStopStartScalingState(cluster.getId(), autoscaleState.getUseStopStartMechanism());
+        Boolean enableStopStartScaling = autoscaleState.getUseStopStartMechanism();
+        if (enableStopStartScaling == null && cluster.getTimeAlerts().isEmpty()) {
+            enableStopStartScaling = shouldEnableStopStartBasedOnEntitlement(cluster);
+        }
```

Comment:
Should this be moved to a separate method too?

Reply:
Done